src/utilities.cpp

```diff
@@ -39,6 +39,21 @@ limitations under the License.
 
 namespace libcellml {
 
+bool canConvertToBasicDouble(const std::string &in)
```

Comment:
Do we really need this separate to the below `convertToDouble` method? Which seems to do exactly the same things but also returns that value...

Reply:
No, it's not the same as `convertToDouble()` since it allows the exponent part (i.e. `123.45` and `1.23e45` are allowed) while `canConvertToBasicDouble()` doesn't (i.e. `123.45` is allowed, but `1.23e45` is not). Notice that the former relies on `isCellMLReal()` while the latter on `isCellMLBasicReal()`.

(When it comes to returning a `bool` of a `double`, I went for `bool` because our current use of `canConvertToBasicDouble()` doesn't care about the value, just whether it can be converted to a basic double, not to mention that it makes branch coverage much simpler.)

Reply:
I would like to see this refactored to reuse the code, this function and `convertToDouble` have similarities that can be combined to reduce repetition.

Reply:
What about you my comment just above?